internal/runners/run/run.go

```diff
@@ -113,16 +99,31 @@ func run(out output.Outputer, subs subshell.SubShell, name string, args []string
 		}
 		subs.SetEnv(env)
 
-		// get the "clean" path (only PATHS that are set by venv)
+		// search the "clean" path first (PATHS that are set by venv)
 		env, err = venv.GetEnv(false, "")
 		if err != nil {
 			return err
 		}
-		path = env["PATH"]
+		path = env["PATH"] + string(os.PathSeparator) + path
```

Comment:
This doesn't seem related to this PR, and reading through you would just end up with PATH being in `path` twice.

Reply:
`GetEnv` is not inheriting the environment so the entry at `env["PATH"]` only contains the cache dir.